orchestration/_tests/test_globus_flow.py

```diff
@@ -183,6 +191,54 @@ def get_run(self, run_id: UUID) -> Dict[str, Any]:
         return self.runs.get(run_id, {})
 
 
+def test_832_decision_flow(mocker: MockFixture):
+    """Test 832 uber decision flow."""
+
+    # Mock the Secret block load using a simple manual mock class
+    class MockSecret:
+        value = str(uuid4())
+
+    mocker.patch('prefect.blocks.system.Secret.load', return_value=MockSecret())
+
+    # Import decision flow after mocking the necessary components
+    from orchestration.flows.bl832.decision_flow import decision_flow
+
+    # Mock read_deployment_by_name with a manually defined mock class
+    class MockDeployment:
+        version = "1.0.0"
+        flow_id = str(uuid4())
+        name = "test_deployment"
+
+    mocker.patch('prefect.client.orchestration.PrefectClient.read_deployment_by_name',
+                 return_value=MockDeployment())
+
+    # Mock run_deployment to avoid executing any Prefect workflows
+    async def mock_run_deployment(*args, **kwargs):
+        return None
+
+    mocker.patch('prefect.deployments.deployments.run_deployment', new=mock_run_deployment)
+
+    # Mock asyncio.gather to avoid actual async task execution
+    async def mock_gather(*args, **kwargs):
+        return [None]
+
+    mocker.patch('asyncio.gather', new=mock_gather)
```

Comment:
Would be nice to have a comment here explaining why we need to stub `asyncio.gather`?